core/dbt/adapters/sql/connections.py

```diff
@@ -128,6 +128,31 @@ def get_result_from_cursor(cls, cursor: Any) -> agate.Table:
 
         return dbt.clients.agate_helper.table_from_data_flat(data, column_names)
 
+    @classmethod
+    def data_type_code_to_name(cls, type_code: Union[int, str]) -> str:
+        """Get the string representation of the data type from the type_code."""
+        # https://peps.python.org/pep-0249/#type-objects
+        raise dbt.exceptions.NotImplementedError(
+            "`data_type_code_to_name` is not implemented for this adapter!"
+        )
+
+    @classmethod
+    def get_column_schema_from_cursor(cls, cursor: Any) -> List[Tuple[str, str]]:
```

Comment:
Are models assessed with this guaranteed to be preexisting or does the empty list indication that it doesn't exist yet?

Reply:
At this point, callers using the method now are guaranteed to have run a query successfully, and yes, an empty `cursor.description` would lead to an empty list result here. I'll add a docstring around the expected behaviour of this method given a cursor.